File: seleniumEasyJava/src/test/java/E2E/steps/progressBar/ProgressBarSteps.java
package E2E.steps.progressBar;

import E2E.runner.TestRunner;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.FluentWait;
import org.openqa.selenium.support.ui.Wait;

import java.time.Duration;
import java.util.function.Function;

import static E2E.runner.TestRunner.driver;

public class ProgressBarSteps {
    @Given("the developer is on the progress bar demo selenium easy website")
    public void theDeveloperIsOnTheProgressBarDemoSeleniumEasyWebsite() {
        driver.get("https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html");
    }

    @When("the developer clicks on the download button")
    public void theDeveloperClicksOnTheDownloadButton() {
        TestRunner.progressBarPage.downloadButton.click();
    }

    @Then("the developer will see one-hundred percentage on the page when the download completes")
    public void theDeveloperWillSeeOneHundredPercentageOnThePageWhenTheDownloadCompletes() {
        boolean percentage = (boolean) TestRunner.fluentWait.until(ExpectedConditions.textToBePresentInElement(
            (TestRunner.progressBarPage.percentageText), "100%"));

        Assert.assertTrue(percentage);
    }
}

File: seleniumEasyPython/steps/progress_bar_steps.py
from behave import Given, When, Then
from selenium.common.exceptions import ElementNotVisibleException, ElementNotSelectableException
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


@Given(u'the developer is on the progress bar demo selenium easy website')
def get_selenium_checkbox_easy_website(context):
    context.driver.get("https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html")


@When(u'the developer clicks on the download button')
def step_impl(context):
    context.progress_bar.download_button().click()


@Then(u'the developer will see one-hundred percentage on the page when the download completes')
def step_impl(context):
    percentage = WebDriverWait(context.driver, 30, 1, [ElementNotVisibleException, ElementNotSelectableException]).\
        until(expected_conditions.text_to_be_present_in_element((By.CLASS_NAME, "percenttext"), "100%"))
    assert percentage
